### PR TITLE
Add support for Committee/Team Senior Members.

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
This is the continuation of #3747. I didn't add anything in this PR to show publicly who are Senior Members as I'm not quite sure how to do that with the current design of our [/about page](https://www.worldcubeassociation.org/about). Maybe it's better to handle this in #3662? If we decide to do that, I'll add this information to the commit message.

I've also updated the helper methods introduced with 3462f62c11fb0c953e7a9b1241cea1f16483972e.

:warning: Final implementation of this feature depends on approval of https://github.com/thewca/wca-documents/pull/34 by WCA Staff and External Staff.

### CODE REVIEW

File: WcaOnRails/spec/features/eligible_voters_spec.rb [hunk 1]
```diff
@@ -20,6 +20,11 @@
       user.team_members.find_by_team_id(wrc_team_id).update!(team_leader: true)
     end
   }
+  let!(:team_senior_member) {
+    FactoryBot.create(:user, :wrc_member).tap do |user|
```
Comment (jfly) [inline WcaOnRails/spec/features/eligible_voters_spec.rb]: Could you add this to the users factory instead? Just like how we do `team_leader` over there. Then you should be able to rewrite this line as:

```ruby
let!(:team_senior_member) { FactoryBot.create(:user, :wrc_member, team_senior_member: true) }
```

File: WcaOnRails/app/models/user.rb [hunk 2]
```diff
@@ -433,24 +434,22 @@ def software_team?
     team_member?(Team.wst)
   end
 
-  # See https://www.worldcubeassociation.org/documents/motions/02.2018.1%20-%20Definitions.pdf
-  def internal_staff?
-    full_delegate? || senior_delegate? || leader_of_any_team? || board_member?
+  def staff?
+    candidate_delegate? || full_delegate? || senior_delegate? || member_of_any_team? || senior_member_of_any_team? || leader_of_any_team? || board_member?
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/user.rb]: Isn't that equivalent to:
```rb
any_kind_of_delegate? || meber_of_any_team? || board_member?
```

File: WcaOnRails/spec/features/eligible_voters_spec.rb (hunk 1 above)
Comment (AlbertoPdRF) [reply WcaOnRails/spec/features/eligible_voters_spec.rb]: Sure!

File: WcaOnRails/app/models/user.rb (hunk 2 above)
Comment (AlbertoPdRF) [reply WcaOnRails/app/models/user.rb]: Indeed, I'll change it!

File: WcaOnRails/spec/features/eligible_voters_spec.rb (hunk 1 above)
Comment (AlbertoPdRF) [reply WcaOnRails/spec/features/eligible_voters_spec.rb]: I'll also change all the other lines that are written like this.

File: WcaOnRails/app/models/user.rb (hunk 2 above)
Comment (operge) [reply WcaOnRails/app/models/user.rb]: If so, then sure.

Comment (AlbertoPdRF) [reply WcaOnRails/app/models/user.rb]: Changed :)

**Comment by AlbertoPdRF** [comment]
This should be safe to be merged now, as https://github.com/thewca/wca-documents/pull/34 has been approved!

**Comment by jonatanklosko** [comment]
Thanks Alberto, I've just deployed the change :ship:

**Pull request merged**